src/Processors/QueryPlan/Optimizations/optimizePrewhere.cpp

```diff
@@ -30,7 +30,7 @@ static void removeFromOutput(ActionsDAG & dag, const std::string name)
 
 void optimizePrewhere(Stack & stack, QueryPlan::Nodes &)
 {
-    if (stack.size() < 3)
+    if (stack.size() < 2)
```

Comment:
why?

Reply:
Because we operate only with `Filter -> Read` chain. I do not know why it was 3 before.
Now it's more important cause child plans for Merge could have only 2 steps at all.